app/src/lib/git/branch.ts

```diff
@@ -65,54 +64,50 @@ export async function deleteLocalBranch(
+  branchTipSha: string
 ): Promise<true> {
-  if (branch.type === BranchType.Local) {
-    await deleteLocalBranch(repository, branch.name)
+  const networkArguments = await gitNetworkArguments(repository, account)
+  const remoteUrl =
+    (await getRemoteURL(repository, remoteName).catch(err => {
+      // If we can't get the URL then it's very unlikely Git will be able to
+      // either and the push will fail. The URL is only used to resolve the
+      // proxy though so it's not critical.
+      log.error(`Could not resolve remote url for remote ${remoteName}`, err)
+      return null
+    })) || getFallbackUrlForProxyResolve(account, repository)
+
+  const args = [...networkArguments, 'push', remoteName, `:${remoteBranchName}`]
+
+  // If the user is not authenticated, the push is going to fail
+  // Let this propagate and leave it to the caller to handle
+  const result = await git(args, repository.path, 'deleteRemoteBranch', {
+    env: await envForRemoteOperation(account, remoteUrl),
+    expectedErrors: new Set<DugiteError>([DugiteError.BranchDeletionFailed]),
+  })
+
+  if (branchTipSha !== undefined) {
+    log.info(`Deleted branch ${remoteBranchName} (was ${branchTipSha})`)
   }
 
-  const remoteName = branch.upstreamRemoteName
-
-  if (includeRemote && remoteName) {
-    const networkArguments = await gitNetworkArguments(repository, account)
-    const remoteUrl =
-      (await getRemoteURL(repository, remoteName).catch(err => {
-        // If we can't get the URL then it's very unlikely Git will be able to
-        // either and the push will fail. The URL is only used to resolve the
-        // proxy though so it's not critical.
-        log.error(`Could not resolve remote url for remote ${remoteName}`, err)
-        return null
-      })) || getFallbackUrlForProxyResolve(account, repository)
-
-    const args = [
-      ...networkArguments,
-      'push',
-      remoteName,
-      `:${branch.nameWithoutRemote}`,
-    ]
-
-    // If the user is not authenticated, the push is going to fail
-    // Let this propagate and leave it to the caller to handle
-    const result = await git(args, repository.path, 'deleteRemoteBranch', {
-      env: await envForRemoteOperation(account, remoteUrl),
-      expectedErrors: new Set<DugiteError>([DugiteError.BranchDeletionFailed]),
-    })
-
-    // It's possible that the delete failed because the ref has already
-    // been deleted on the remote. If we identify that specific
-    // error we can safely remote our remote ref which is what would
-    // happen if the push didn't fail.
-    if (result.gitError === DugiteError.BranchDeletionFailed) {
-      const ref = `refs/remotes/${remoteName}/${branch.nameWithoutRemote}`
-      await deleteRef(repository, ref)
-    }
+  // It's possible that the delete failed because the ref has already
+  // been deleted on the remote. If we identify that specific
+  // error we can safely remote our remote ref which is what would
```

Comment:
Nit: this is not your typo, but is still a typo 😂 
```suggestion
  // error we can safely remove our remote ref which is what would
```